src/platform/cert_capi.c

```diff
@@ -700,6 +710,208 @@ CxPlatCertParseChain(
     return (QUIC_CERTIFICATE*)LeafCertCtx;
 }
 
+_Success_(return != 0)
+QUIC_STATUS
+CxPlatGetPortableCertificateFromSerialized(
+    _In_ QUIC_CERTIFICATE* SerializedCertificate,
+    _Out_ QUIC_PORTABLE_CERTIFICATE* PortableCertificate
+    )
+{
+    QUIC_STATUS Status;
+    DWORD LastError;
+    HCERTSTORE TempCertStore = NULL;
+    CERT_BLOB* SerializedCertificateStore = (CERT_BLOB*)SerializedCertificate;
+    CERT_BLOB Blob = {0};
+    PCCERT_CONTEXT CurrentCertContext = NULL;
+    PCCERT_CONTEXT LeafCertContext = NULL;
+
+    PortableCertificate->PlatformCertificate = NULL;
+
+    HCERTSTORE SerializedStore =
+        CertOpenStore(
+            CERT_STORE_PROV_SERIALIZED,
+            X509_ASN_ENCODING,
+            0,
+            CERT_STORE_DEFER_CLOSE_UNTIL_LAST_FREE_FLAG,
+            SerializedCertificateStore);
+    if (SerializedStore == NULL) {
+        LastError = GetLastError();
+        if (LastError == ERROR_OUTOFMEMORY ||
+            LastError == ERROR_NOT_ENOUGH_MEMORY) {
```

Comment:
Why either?

Reply:
I copied this from internal, so I assume somebody found CertOpenStore could return a different error sometimes.